lib/rust/parser/src/syntax/tree.rs

```diff
@@ -194,11 +230,140 @@ impl<'s> span::Builder<'s> for Error {
 }
 
 
+// == Code Blocks ===
+
+/// A line of code.
+#[derive(Debug, Clone, PartialEq, Eq, Visitor, Reflect, Serialize, Deserialize)]
+pub struct Line<'s> {
+    /// The content of the line, if any.
+    pub expression: Option<Tree<'s>>,
+    /// The end-of-line token.
+    pub newline:    token::Newline<'s>,
+}
```

Comment:
It's widely used standard that in parsers we remember the left spacing and left newline characters, not right ones. It would also make our life easier (e.g. consider a block that ends at the end of file - you dont have res enplane then).

Reply:
Either approach will work for us--in my design doc, I wrote that we can treat EOF as a zero-length newline, so that we handle files the same way whether or not they end in a newline character. We could instead insert a zero-length newline at the beginning of the file. However, that approach would be unconventional. It is common to treat a file that doesn't end with a newline as a special case. I have never encountered any format or application in which every line is expected to begin with a newline. Every editor I've ever used shows a file that starts with a newline as a file with a blank first line, because the universal standard is that a newline ends a line, rather than marking its start.